packages/sit-onyx/src/components/OnyxDataGrid/features/index.ts

```diff
@@ -0,0 +1,178 @@
+import { h, type Component, type WatchSource } from "vue";
+import type { DataGridRendererColumn, DataGridRendererRow } from "../../..";
+import type { DataGridEntry, DataGridMetadata } from "../types";
+
+export type TableFeature<TEntry extends DataGridEntry, TFeatureName extends symbol> = {
+  /**
+   * Unique name and identifier of the table feature
+   */
+  name: TFeatureName;
+
+  /**
+   * An array of reactive states that should trigger a table re-generation
+   */
+  watch: WatchSource[];
+
+  /**
+   * Allows modifying the table state as a whole.
+   */
+  mutation?: {
+    func: (state: TEntry[]) => void;
+  };
+
+  /**
+   * Allows the modification of the headers.
+   */
+  header?: {
+    /**
+     * Adds header icon button(s).
+     * `iconComponent` of an action is shown after the header label.
+     * The components must be ARIA-conform buttons.
+     */
+    actions?: (column: keyof TEntry) => {
```

Comment:
Imho the name `actions` can be confusing since there might also be actions before the header, not only after it.
What do you think about making it flexible here and provide a `before` as well as `after` property where any components can be passed? Or we add a property `position: "before" | "after"` to each array entry

Reply:
Yes, we can certainly do that when we have requirements for it.